src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ELEMENTS = 10;
+    private static final int INCREMENT_PERCENTS = 50;
+
+    private int size;
+    private Object[] list;
+
+    public ArrayList() {
+        size = 0;
+        list = new Object[MAX_ELEMENTS];
+    }
+
     @Override
     public void add(T value) {
-
+        checkingListForCompl();
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        checkingListForCompl();
+        Object[] newArray = Arrays.copyOfRange(list, index, size);
+        list = Arrays.copyOf(Arrays.copyOfRange(list, 0, index), list.length);
+        size = index + 1;
+        for (Object o : newArray) {
+            list[size] = o;
+            size++;
+        }
+        list[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > this.list.length) {
+            checkingListForCompl(size);
+        }
+        if (!list.isEmpty()) {
+            for (int i = 0; i < list.size(); i++) {
+                this.list[size] = list.get(i);
+                size++;
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Can`t find element by index" + index);
+        }
+        return (T)list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Can`t find element by index" + index);
+        }
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        Object el = list[index];
```

Comment:
use full names for valieables
```suggestion
        Object element = list[index];
```